### PR TITLE
BenchmarkResult class introduced

### PR DESCRIPTION
**Pull request by jakethekoenig**
Miscellaneous Changes
* `session_context` added to `session` and used to access transcripts and costs.
* cost added to benchmark dashboard.

Todo:
* [ ] Use same BenchmarkResult in benchmark_runner.
* [ ] Make BenchmarkResultSummary class.
* [ ] Make dashboard for benchmark_runner.

### CODE REVIEW

File: tests/benchmarks/benchmark_result.py [hunk 1]
```diff
@@ -0,0 +1,29 @@
+import json
+from typing import Optional
+
+import attr
+
+from mentat.transcripts import Transcript
+
+
+@attr.define
+class BenchmarkResult:
```
Comment (PCSwingle) [inline tests/benchmarks/benchmark_result.py:10]: I would've preferred to make this a TypedDict if 3.11 was our minimum version (NotRequired was added), but for 3.10 support I think this (with to_json and from_json) is definitely the best call.

**Review by PCSwingle — APPROVED** [review]
Thanks for adding this!

**Comment by jakethekoenig** [comment]
> Thanks for adding this!

I was going to add the todos to this PR but since you've approved it and this change sort of stands alone I'll go ahead and merge.

**Comment by PCSwingle** [comment]
> > Thanks for adding this!
> 
> I was going to add the todos to this PR but since you've approved it and this change sort of stands alone I'll go ahead and merge.

Oh yeah I somehow completely missed the todo's when I reviewed this; either way, this is a good modular change so I'm happy with having a small PR like this.

**Pull request merged**